### ISSUE TITLE
v2.0 fails to build for macOS

### ISSUE THREAD

**Issue by msfjarvis** [Genymobile/scrcpy#3809]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: macOS Ventura 13.2.1
 - scrcpy version: 2.0
 - installation method: manual build
 - device model: Not applicable
 - Android version: Not applicable

**Describe the bug**

I am a maintainer for the scrcpy package in [Nixpkgs](https://github.com/nixos/nixpkgs).

While [trying to upgrade](https://github.com/NixOS/nixpkgs/pull/220783), the v2.0 release fails to build on macOS with clang 11 due to errors from the `bool` type template not expanding correctly (as I can gather, my C knowledge is basically non-existent).

```
../app/src/screen.c:821:13: error: expected expression
            bool ok = sc_screen_init_size(screen);
            ^
/nix/store/dkgjv2srx116b0f9qnp9vcy3jjk5f39q-clang-wrapper-11.1.0/resource-root/include/stdbool.h:15:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
../app/src/screen.c:822:18: error: use of undeclared identifier 'ok'
            if (!ok) {
                 ^
1 warning and 2 errors generated.
ninja: build stopped: subcommand failed.
```

At the moment I plan to mark the package as broken for macOS and only support Linux, but it'd be great to have this fixed.

**Comment by rom1v** [comment]
This was fixed by #3787. Use `master` (or at least 6b769675fa68e60c9765022e43c4d7b1e329353a).

**Comment by msfjarvis** [comment]
Oh, I'll pull in the patch as a stopgap, thanks!.
